feat: add `math/base/special/kernel-sincosf`

anandkaranubc opened the pull request:
Progresses https://github.com/stdlib-js/stdlib/issues/649

## Description

> What is the purpose of this pull request?

This pull request:

-  Adds the complete implementation for `math/base/special/kernel-sincosf`

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   Progresses https://github.com/stdlib-js/stdlib/issues/649

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

anandkaranubc committed:
feat: add `math/base/special/kernel-sincosf`

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: passed
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: passed
  - task: lint_c_benchmarks
    status: passed
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: passed
  - task: lint_license_headers
    status: passed
---

Planeshifter commented on lib/node_modules/@stdlib/math/base/special/kernel-sincosf/lib/index.js:
```diff
@@ -0,0 +1,59 @@
+* @example
+* var kernelSincosf = require( '@stdlib/math/base/special/kernel-sincosf' );
+*
+* var v = kernelSincosf( 0.0, [ 0.0, 0.0 ], 1, 0 );
+* // returns [ ~0.0, ~1.0 ]
```
```suggestion
* var v = kernelSincosf( 0.0 );
* // returns [ ~0.0, ~1.0 ]
```

Planeshifter commented on lib/node_modules/@stdlib/math/base/special/kernel-sincosf/README.md:
````diff
@@ -0,0 +1,210 @@
+// returns [ NaN, NaN ]
+```
+
+#### kernelSincosf( x, out, stride, offset )
````
This function signature is for the assign method. Need to update the heading and also the below example code; it would be good to change it so it is not simply equivalent to code block L37-L49, e.g. by demonstrating `out` array being mutated and/or by using different strides/offsets.

anandkaranubc replied:
Thanks @Planeshifter for the feedback! I have followed the same what was implemented in `modff`.

Sorry for the oversight...

Planeshifter commented on lib/node_modules/@stdlib/math/base/special/kernel-sincosf/README.md:
```diff
@@ -0,0 +1,210 @@
+var y;
+var i;
+for ( i = 0; i < x.length; i++ ) {
+    y = kernelSincosf( x[ i ], [ 0.0, 0.0 ], 1, 0 );
```
```suggestion
    y = kernelSincosf( x[ i ] );
```

Planeshifter commented on lib/node_modules/@stdlib/math/base/special/kernel-sincosf/README.md:
````diff
@@ -0,0 +1,210 @@
+-   **cosine**: `[out] float*` destination for the cosine.
+
+```c
+void stdlib_base_kernel_sincosf( const double x, const double y, float* sine, float* cosine );
````
Copy & paste error from `kernel-sincos`:

```suggestion
void stdlib_base_kernel_sincosf( const double x, float* sine, float* cosine );
```

Planeshifter commented on lib/node_modules/@stdlib/math/base/special/kernel-sincosf/examples/index.js:
```diff
@@ -0,0 +1,32 @@
+var y;
+var i;
+for ( i = 0; i < x.length; i++ ) {
+	y = kernelSincosf( x[ i ], [ 0.0, 0.0 ], 1, 0 );
```
```suggestion
	y = kernelSincosf( x[ i ] );
```

Planeshifter reviewed (changes requested):
Looks good overall, but there are a few issues carried over from `kernel-sincos`. Aside from addressing them here, it would be great if you could also update the double-precision variant, please.

anandkaranubc committed:
Merge branch 'develop' into feat/kernel-sincosf

anandkaranubc committed:
chore: apply suggestions from review

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

Planeshifter reviewed (approved):
LGTM after the edits. Thanks!
